service_configuration_lib/cached_view.py

```diff
@@ -77,16 +87,30 @@ def log_inotify_constants() -> None:
         log.info(f'max_user_instances: {pyinotify.max_user_instances.value}')
         log.info(f'max_user_watches: {pyinotify.max_user_watches.value}')
 
-    def process_events(self) -> None:
+    def process_events(self, limit: int = DEFAULT_PROCESS_EVENTS_LIMIT) -> None:
         """This method should be periodically invoked to process queued inotify
         events.  See the class docstring for more context.
+        Args:
+        :param limit: Optional, rough limit of events to process per call
         """
-        while self._notifier.check_events(timeout=0):
+        before_processed_count = self._processed_events_count
+        currently_processed = 0
+        while currently_processed >= 0 and currently_processed < limit and self._notifier.check_events(timeout=0):
```

Comment:
is it possible for `currently_processed` to become negative? If so, when (and can we test that case 😄)?

Reply:
Yes, I added tests